### ISSUE TITLE
Bug Report: when playing a song with speed lower then one it randomly stops the song 

### ISSUE THREAD

**Issue by DeceptedPar** [KadeArchive/Kade-Engine#1890]
### If you are playing a downloaded version of the game, what operating system are you using?

Windows (x86_64)
## What version of the game are you using? Look in the bottom left corner of the main menu.
0.2.7.1
## Have you identified any steps to reproduce the bug? If so, please describe them below. Use images if possible.
having the speed rate in freeplay lower than one
for this example I used milf at 0.7
![image](https://user-images.githubusercontent.com/64243677/130362479-bc10ad0c-11e6-4e0a-93f2-889784fb4b01.png)
then a little bit past the halfway mark of the song it just stops
## Please describe your issue. Provide extensive detail and images if possible.
basically, this will work on ANY song I just used milf as an example, but the issue is, you can't fully complete the song when the speed is lower then 1.0 due to the song stopping where the song would stop if it had 1x speed

if this doesn't help, I'm sorry. I seriously am not the best at explaining

**Comment by BoloVEVO** [comment]
Already did a pull request to fix that. If you want you can copy the code. I'm not sure if it works in 1.7 but in 1.8+ it does! 

https://github.com/KadeDev/Kade-Engine/pull/2858
